horace_core/sqw/file_io/@faccess_sqw_v3_3/faccess_sqw_v3_3.m

```diff
@@ -163,16 +163,29 @@
             obj.position_info_pos_= obj.instr_sample_end_pos_;
             %
             data = obj.extract_correct_subobj('data');
-            obj.pix_range_ = data.pix.pix_range;            
-            num_pix = data.pix.num_pixels;            
-            
+            obj.pix_range_ = data.pix.pix_range;
+            num_pix = obj.npixels;
+
             if any(any(obj.pix_range_ == PixelData.EMPTY_RANGE_)) && num_pix > 0
+                hc = hor_config;
+                buf_size = hc.mem_page_chunk_size_byte_conversion;
+                pix_page = hc.pixel_page_size;
+                ll = hc.log_level;
+                if 2*buf_size < pix_page
+                    clOb = onCleanup(@()set(hc,'pixel_page_size',pix_page));
+                    % set up pixel_page size of to be buf_size converted in bytes
+                    hc.mem_page_chunk_size_byte_conversion = hc.mem_chunk_size;
+                end
+                if ll>0
+                    fprintf(...
+                        '*** Recalculating pixel range to upgrade file format to a latest binary version\n')
```

Comment:
To "a" latest binary version?

Reply:
yes, here I do not know what the latest version is but all subsequent should need this.  May be too much information for user to swallow, but "a" here was intentional.  May be indeed too many info and the would be better